### DISCUSSION TITLE
Debug Adapter Protocol integration

### DISCUSSION THREAD

**Discussion by fzalila** [eclipse-langium/langium#262] [Q&A] [answered]
Hi langium, 

First of all, thank you for this amazing project. 
As langium supports LSP, do you plan to support Debug Adapter Protocol (DAP) in langium to easily implement and/or generate a debugger to a DSL created with langium? 

Thanks in advance, 

Faiez

**Comment by msujew** [comment, marked as answer]
Hi @fzalila, thank you for your question.

First off, as our main focus for the next few months will be to reach a stable version of Langium and implement the main features we need for DSL support as well as LSP support, integration for other protocols/framework won't be our top priority. Nevertheless, it's still good to keep these topics in mind.

One other approach (compared to DAP) that we had in mind regarding debuggability was using source maps. As they are language independent, they can always be used to map generated source code for JS for example back to the original DSL. That would allow for quite easy debugging support of generated code.

In the longer term we would probably try to provide infrastructure for building interpreters/compilers for DSL written in Langium, which is where the DAP support would also come in play.

**Comment by fzalila** [comment]
Thank you @msujew for the quick reply! 
Very interesting strategy!  
I am looking forward to see how this project progresses!
